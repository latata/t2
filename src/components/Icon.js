import React, { Component } from 'react';

class Icon extends Component {
  render() {
    const classNames = `icon ${this.props.className}`;

    return (<svg className={classNames}>
      <use xlinkHref={`#${this.props.name}`} className={`icon-${this.props.name}`} />
            </svg>);
  }
}

export default Icon;
