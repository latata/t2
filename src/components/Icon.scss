.icon {
  width: 16px;
  height: 16px;
  fill: currentColor;
}

.btn {
  &.with-label .icon {
    margin-right: 6px;
  }

  .icon {
    vertical-align: middle;
    margin-top: -2px;
  }
}
