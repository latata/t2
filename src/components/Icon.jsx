import React from 'react';
import PropTypes from 'prop-types';
import './Icon.css';

export default function Icon({ className, name }) {
  let classNames = 'icon';

  if (className) {
    classNames += ` ${className}`;
  }

  return (
    <svg className={classNames}>
      <use xlinkHref={`#${name}`} className={`icon-${name}`} />
    </svg>);
}

Icon.propTypes = {
  name: PropTypes.string.isRequired,
  className: PropTypes.string,
};
