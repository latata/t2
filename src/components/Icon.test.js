// Link.react.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import Icon from './Icon';

test('Icon without className', () => {
  const component = renderer.create(
    <Icon name="test" />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Icon with className', () => {
  const component = renderer.create(
    <Icon name="test" className="test-classname" />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Icon with multiple className', () => {
  const component = renderer.create(
    <Icon name="test" className="test-classname second-class third-class" />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
