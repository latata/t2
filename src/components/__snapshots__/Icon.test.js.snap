// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon with className 1`] = `
<svg
  className="icon test-classname"
>
  <use
    className="icon-test"
    xlinkHref="#test"
  />
</svg>
`;

exports[`Icon with multiple className 1`] = `
<svg
  className="icon test-classname second-class third-class"
>
  <use
    className="icon-test"
    xlinkHref="#test"
  />
</svg>
`;

exports[`Icon without className 1`] = `
<svg
  className="icon"
>
  <use
    className="icon-test"
    xlinkHref="#test"
  />
</svg>
`;
